### PR TITLE
SJRK-370: Updating to latest dev release of Infusion.

### PR DESCRIPTION
**Pull request by jobara**
Using the [FLUID-6145](https://github.com/fluid-project/infusion/pull/972) line.

https://issues.fluidproject.org/browse/SJRK-370

### CODE REVIEW

**Comment by jobara** [comment]
@BlueSlug could you please verify that this PR is working and nothing has broken?

**Comment by amb26** [comment]
3.0.0-dev.20200720T221751Z.573b5f3c5.FLUID-6145 is now ready

**Comment by jobara** [comment]
> 3.0.0-dev.20200720T221751Z.573b5f3c5.FLUID-6145 is now ready

Thanks @amb26

**Comment by jobara** [comment]
@BlueSlug I've updated all of the dependencies in this PR. Could you please review and make sure nothing has broken? Also it would be good to review the dependencies in the package.json file to make sure that they are all still needed and that they are in the correct section (dependencies vs devDependencies).

**Comment by BlueSlug** [comment]
Things are mostly looking good, and I can confirm that FLUID-6512 and FLUID-6523 have been solved (particular quirks in the Selection Reader due to Storytelling Tool-specific markup notwithstanding).

There are still, however, a couple of issues:
- `testServerWithStorage.js:18` should read `var { v1: uuidv1 } = require("uuid");` rather than `uuidv1 = require("uuid/v1");` (and the comma on the line above should become a semicolon, accordingly)
- the Orator doesn't seem to be working at all in Edge 44


The error I was receiving for the Orator failure is: `Unhandled promise rejection: [object SpeechSynthesisEvent]`
and a subsequent attempt returns: `ASSERTION FAILED: Error: resolving promise [object Object] which has already received "reject"`. It alternates between those two messages if I continue to attempt to play the Orator.

I also reviewed the dependencies. All of them are currently required and correctly placed EXCEPT the following:
- `chance`: this is only used in storyFabricator.js in order to generate fake stories in bulk. I've used it twice ever, but if we'd like to remove this, it should be tackled in a separate Jira
- `handlebars`: this can be removed, though there must still be references to its files on each HTML page since it's used by `fluid-handlebars`
- `eslint-plugin-markdown`: I believe it is not needed. I think `fluid-grunt-lint-all` brings this in as a dependency of its own
- `testem`: this is not needed, though it will be added by SJRK-122, so I don't feel strongly about its continued inclusion at this point

**Comment by jobara** [comment]
> I also reviewed the dependencies. All of them are currently required and correctly placed EXCEPT the following:
> 
> * `chance`: this is only used in storyFabricator.js in order to generate fake stories in bulk. I've used it twice ever, but if we'd like to remove this, it should be tackled in a separate Jira
> * `handlebars`: this can be removed, though there must still be references to its files on each HTML page since it's used by `fluid-handlebars`
> * `eslint-plugin-markdown`: I believe it is not needed. I think `fluid-grunt-lint-all` brings this in as a dependency of its own
> * `testem`: this is not needed, though it will be added by SJRK-122, so I don't feel strongly about its continued inclusion at this point

I'll just leave the dependencies as is for now. The `eslint-plugin-markdown` is needed for `fluid-grunt-lint-all` (see: fluid-project/fluid-grunt-lint-all#13). I'm afraid a similar issue might happen for handlebars too. We will need to investigate that.

**Comment by jobara** [comment]
> * `testServerWithStorage.js:18` should read `var { v1: uuidv1 } = require("uuid");` rather than `uuidv1 = require("uuid/v1");` (and the comma on the line above should become a semicolon, accordingly)

Fixed

**Comment by jobara** [comment]
> * the Orator doesn't seem to be working at all in Edge 44

There is a bug in the Orator component in that it didn't completely handle errors. In the FLUID-6145 line promises are required to handle rejections. I've filed this issue as [FLUID-6533](https://issues.fluidproject.org/browse/FLUID-6533) and addressed in fluid-project/infusion#1004. However, that doesn't address what was causing the error. In MS Edge 44 (non-chromium Edge) it seems that the the SpeechSynthesisUtterance object is specified with a lang for which there is no corresponding synthesizer, it throws an error. Other browsers will find the most appropriate synthesizer to use. For example in the Orator demo the lang is set to "en-CA". In my Windows 10 VM, MS Edge 44 only has synthesizers that support "en-US". If the lang is even specified as "en" it will still throw an error. This seems like a bug in the browser and it's unlikely it'll get fixed now that they've moved to a new version of Edge that is based off of Chromium. I've filed this as [FLUID-6535](https://issues.fluidproject.org/browse/FLUID-6535) although I'm not sure we'll fix it as this is no longer a supported browser for Infusion.

**Comment by BlueSlug** [comment]
I approve this for merging :)

**Comment by jobara** [comment]
@cindyli this PR is ready for review.

File: package.json [hunk 1]
```diff
@@ -17,29 +17,31 @@
...
-        "fs-extra": "7.0.1",
-        "gpii-binder": "1.1.0",
-        "gpii-handlebars": "2.1.0",
-        "gpii-location-bar-relay": "1.0.5-dev.20200128T132941Z.86c73d0.GPII-2131",
-        "handlebars": "4.7.2",
-        "infusion": "3.0.0-dev.20191219T155607Z.32876d29c.FLUID-6145",
-        "jpeg-autorotate": "5.0.2",
+        "fluid-handlebars": "2.1.3",
+        "fs-extra": "9.0.1",
+        "gpii-location-bar-relay": "1.0.5",
```
Comment (cindyli) [inline package.json]: `gpii-location-bar-relay` no longer exists on [npmjs](https://www.npmjs.com/search?q=gpii-location-bar-relay). It looks like having been replaced by [`fluid-local-bar-relay`](https://www.npmjs.com/package/fluid-location-bar-relay).

**Comment by cindyli** [comment]
If you search `gpii` across the code base, `gpii-express` appears at several places. Since it is not a dependency or devDependency of this repo, those places can be cleaned up.

`gpii-location-bar-relay` shows up at several places too. They should be fixed to `fluid-location-bar-relay` as what's mentioned in the previous comment.

**Comment by BlueSlug** [comment]
I've tested the tool after @jobara's latest updates, and:
- the functionality handled by `fluid-location-bar-relay` is working as expected in Chrome 84, Firefox 78 and Edge 44
- the browser tests are passing in all three browsers
- the server tests are passing

**Comment by jobara** [comment]
@cindyli this PR is ready for more review.

**Comment by cindyli** [comment]
Please update `package-lock.json` too. 

At the moment, running `npm ci` throws this error:

```
npm ERR! cipm can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
```

**Comment by cindyli** [comment]
> Please update `package-lock.json` too.
>

Sorry, my bad. `package-lock.json` has never been pushed up in this repo. It's just me who needs to remove my local copy.

**Pull request merged**